ocaml/xe-cli/newcli.ml

```diff
@@ -293,7 +300,16 @@ let handle_unmarshal_failure ex ifd = match ex with
     else raise e
   | e -> raise e
 
-let main_loop ifd ofd =
+let assert_upload_filename_in_args filename permitted_filenames =
+  (* Prefix match instead of exact match is used here to workaround the old xva format that request files
+   * are not in the command line *)
+  let is_filename_permited = List.exists (fun file_in_arg -> Astring.String.is_prefix ~affix:file_in_arg filename)
```

Comment:
This is not necessarily enough, you also want to block `/../` after the prefix, or check the canonicalized path.

Reply:
Yes, I logged this and we can refine it during the more restrict testing. Thanks.

Reply:
Also I think similar filtering should be applied for download, not just upload.

Reply:
IIRC there is a discussion about the whole name-checking instead of prefix, since now we have the old style of XVA. after remove the old style of XVA, we will change the checking here I think.

Reply:
See fileserver.ml, there is already a helper: Stdext.Unixext.resolve_dot_and_dotdot.

Reply:
Will log these places you point out. Thank you.